### PR TITLE
fix(table): apply lightweight styling to header

### LINKED ISSUE #837: [Table] Clean up lightweight design
### Describe the bug

Table `lightweight` prop no longer affects header styling.

### PR DESCRIPTION
**Pull request by bryancboyd**
Closes #837 

**Summary**

- Fixes lightweight table header styling

image

image

image

image

**Acceptance Test (how to verify the PR)**

- Verify **with lightweight design** story.  In a stateful store, toggle the `lightweight` knob and verify the column selection, filtering, and bulk action states.

### CODE REVIEW

File: src/components/Table/TableHead/_table-head.scss [hunk 1]
```diff
@@ -2,10 +2,14 @@
 @import '~carbon-components/scss/globals/scss/layout';
 
 .#{$prefix}--data-table {
```
Comment (davidicus) [inline src/components/Table/TableHead/_table-head.scss]: Does this need to be scoped to only be applied when lightweight  class is applied?

**Comment by davidicus** [comment]
I think the headers have an issue. This is from another story with the lightweight knob turned on. 
image

**Comment by davidicus** [comment]
@bryancboyd any changes?

**Comment by bryancboyd** [comment]
@davidicus working this now

**Pull request merged**